File: solutions/java/src/restaurantmanagementsystem/state/OrderedState.java
package restaurantmanagementsystem.state;

import restaurantmanagementsystem.model.OrderItem;

public class OrderedState implements OrderItemState {
    @Override
    public void next(OrderItem item) {
        item.setState(new PreparingState());
    }

    @Override
    public void prev(OrderItem item) {
        System.out.println("This is the initial state.");
    }

    @Override
    public String getStatus() { return "ORDERED"; }
}

File: solutions/csharp/restaurantmanagementsystem/States/OrderedState.cs
class OrderedState : IOrderItemState
{
    public void Next(OrderItem item)
    {
        item.SetState(new PreparingState());
    }

    public void Prev(OrderItem item)
    {
        Console.WriteLine("This is the initial state.");
    }

    public string GetStatus() => "ORDERED";
}